scripts/addons.py

```diff
@@ -0,0 +1,225 @@
+            if self._check_is_url(_args[1]):
+                # repo OR organization/repo AND url AND branch
+                self._parse_organization_repo(_args[0])
+                self._parse_url(_args[1])
+                self._set_branch(_args[2])
+                self.short_dependent = _args[0]
+
+    @property
+    def path(self):
+        return '%s%s' % (EXTRA_ADDONS_PATH, self.short_dependent)
+
+    @property
+    def resolve_url(self):
+        _resolve_url = '%s%s/%s/%s.git' % (
+            self.scheme,
+            self.git_repo_host,
+            self.organization,
+            self.repository
+        )
+        return _resolve_url
+
+    @property
+    def download_cmd(self):
+        if self.branch:
+            cmd = 'git clone -b %s %s %s' % (
+                self.branch, self.resolve_url, self.path
+            )
+        else:
+            cmd = 'git clone %s %s' % (
+                self.resolve_url, self.path
+            )
+        return cmd.split()
+
+    @property
+    def update_cmd(self):
+        if self.branch:
+            cmd = 'git --git-dir=%s/.git --work-tree=%s pull origin %s' % (
+                self.path, self.path, self.branch
+            )
+            return cmd.split()
+        else:
+            branch_cmd = 'git --git-dir=%s/.git --work-tree=%s branch' % (
+                self.path, self.path
+            )
+            output = check_output(branch_cmd, shell=True)
+            for line in output.split('\n'):
+                if line.startswith('*'):
+                    self._set_branch(line.replace('* ', ''))
+                    cmd = 'git --git-dir=%s/.git --work-tree=%s pull origin %s' % (
+                        self.path, self.path, self.branch
+                    )
+                    return cmd.split()
+
+    def download(self):
+        if self.path in ADDONS_PATH:
+            return
+        if os.path.exists(self.path):
+            print('PULL: %s %s' % (self.path, self.branch))
+            cmd = self.update_cmd
+            call(cmd)
+        else:
+            print('CLONE: %s %s' % (self.path, self.branch))
+            call(self.download_cmd)
+            # TODO update branch if it's empty
+        ADDONS_PATH.append(self.path)
+        self.download_dependency()
+
+    def download_dependency(self):
+        filename = '%s/oca_dependencies.txt' % self.path
+        if not os.path.exists(filename):
+            return
+        repo_list = []
+        with open(filename) as f:
+            for line in f:
+                l = line.strip('\n').strip()
+                if l.startswith('#') or not l:
+                    continue
+                repo_list.append(Repo(l, self))
+        f.close()
+        for repo in repo_list:
+            repo.download()
+
+
+def write_addons_path():
+    with open(ODOO_CONF, 'a') as f:
+        f.write('addons_path = %s' % ','.join(list(set(ADDONS_PATH))))
+    f.close()
+
+
+def main():
+    dependent = os.environ.get('ADDONS_REPO')
```

Comment:
What do you mean by `dependent`? That there's a repo defined as ENV variable in the d-c.yml?